ci/sandbox/deploy-pipeline.yaml

```diff
@@ -0,0 +1,165 @@
+---
+apiVersion: concourse.k8s.io/v1beta1
+kind: Pipeline
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: deploy
+spec:
+  exposed: true
+  pipelineString: |
+
+    github_source: &github_source
+      uri: https://github.com/alphagov/verify-metadata-controller.git
+      organization: alphagov
+      owner: alphagov
+      repository: verify-metadata-controller
+      github_api_token: ((github.api-token))
+      access_token: ((github.api-token))
+      approvers: ((trusted-developers.github-accounts))
+      required_approval_count: 2
+      commit_verification_keys: ((trusted-developers.gpg-keys))
+
+    harbor_source: &harbor_source
+      username: ((harbor.harbor_username))
+      password: ((harbor.harbor_password))
+      harbor:
+        url: ((harbor.harbor_url))
+        prevent_vul: "false"
+      notary:
+        url: ((harbor.notary_url))
+        root_key: ((harbor.root_key))
+        delegate_key: ((harbor.ci_key))
+        passphrase:
+          root: ((harbor.notary_root_passphrase))
+          snapshot: ((harbor.notary_snapshot_passphrase))
+          targets: ((harbor.notary_targets_passphrase))
+          delegation: ((harbor.notary_delegation_passphrase))
+
+    task_toolbox: &task_toolbox
+      type: docker-image
+      source:
+        repository: govsvc/task-toolbox
+        tag: 1.5.0
+
+    resource_types:
+
+    - name: github
+      type: registry-image
+      source:
+        repository: govsvc/concourse-github-resource
+        tag: v0.0.1
+
+    - name: harbor
+      type: docker-image
+      privileged: true
+      source:
+        repository: govsvc/gsp-harbor-docker-image-resource
+        tag: 0.0.1553882420
+
+    resources:
+
+    - name: release
+      type: harbor
+      icon: tag
+      source:
+        <<: *harbor_source
+        repository: registry.((cluster.domain))/eidas/vmc-release
+
+    jobs:
+
+    - name: deploy
+      serial: true
+      plan:
+
+      - get: release
+        trigger: true
+
+      - task: extract-release
+        image: release
+        config:
+          platform: linux
+          outputs:
+          - name: release
+          run:
+            path: /bin/sh
+            args:
+            - -euc
+            - |
+              echo "Extracting chart package from the release image..."
+              cp -r /vmc-chart-package/* release/
```

Comment:
Isn't the release package in a tar file, `rootfs.tar`, at this point?
Sorry if it's a stupid question, i'm trying to follow these pipelines through properly for the first time.

Reply:
No problem, it is not very intuitive to understand. Basically, I'm abusing Docker to store a bunch of files. Yes an image is just a tar file but it also has a lot of metadata and junk when you just un-zip it. So the solution is to `docker run` that image and extract the files you need to Concourse.
This is done here:
https://github.com/alphagov/verify-metadata-controller/blob/5e63e15738876565275152ebeb8d098c4704a498/ci/sandbox/deploy-pipeline.yaml#L78-L79
The commands run in the container created from the `release` image which contains the files we want to extract

Reply:
Still confused I'm afraid. In this release pipeline, we put the release image back up to harbour with the package in a tar called `rootfs.tar`, here: https://github.com/alphagov/verify-metadata-controller/pull/46/files#diff-babd19c377dfb4915ff15a0d30d2d787R245

Then in the deploy pipeline we get that image and just run `cp -r /vmc-chart-package/* release/`. At what point does that `rootfs.tar` get un-tarred back into `/vmc-chart-package`?

Reply:
Here:
```
  - task: extract-release 
         image: release 
```
This internally does `docker run ...` so Docker un-tars the tar and mounts the file system. Then we can just access the files we need. We could just untar the tar ourselves but then we'd have to deal with all the junk added by Docker. Basically, we use docker to wrap the release files and it's best to also use it to unwrap them.

Reply:
Ah ok, so `rootfs.tar` is the containers actual file system. Got it. I guess the clue's in the name.

Reply:
The tar actually contains the file system wrapped in a lot of Docker junk/metadata so it's buried in a file structure created by docker. But essentially yes.